lib/Agrammon/OutputFormatter/Excel.pm6

```diff
@@ -84,6 +100,22 @@ sub output-as-excel(
     return $workbook;
 }
 
+sub render-filters($sheet, $row is rw, $col, $model, Agrammon::Outputs::FilterGroupCollection $collection,
+                   $unit, $language, Bool :$all-filters) {
+    my @results = $collection.results-by-filter-group(:all($all-filters));
+    for @results {
+        my %keyFilters := .key;
+        my @filters = translate-filter-keys($model, %keyFilters).map: -> $trans { %( label => $trans.key, enum => $trans.value ) };
+        my $value := .value;
+        for @filters -> $filter {
+            $sheet.set($row, $col+2, $filter<enum>{$language}, :horizontal-align(RightAlign));
```

Comment:
We don't need to build up the throw-away hash here; in the JSON one its there to get us over the JSON boundary. Here we can just do something like `my %filters := translate-filter-keys($model, %keyFilters);` and then `for %filters.kv -> %label, %enum { }`.

Reply:
Fixed (left comment there on purpose).